### DISCUSSION TITLE
Multiple materials in same geometry (from GMSH)

### DISCUSSION THREAD

**Discussion by ndaman** [idaholab/moose#16247] [Q&A Meshing] [answered]
I am trying to assign to different material properties to named blocks (using the Tensor Mechanics module). I have no trouble defining the two different blocks in GMSH, but if I do not add "coherence" in GMSH, then the two blocks are not joined together in MOOSE. When I do add "coherence" then MOOSE gives this error: "I don't believe that we can specify multiple subdomain or boundary ids for a given geometric entity"

What is the correct way to give sub-sections of the same mesh different material properties?

**Comment by ndaman** [comment]
I thought I would try going about this another way, by removing the "coherence" (union) in GMSH the two geometric entities are separate, I thought that I could then apply an [EqualValueEmbeddedConstraint](https://mooseframework.inl.gov/source/constraints/EqualValueEmbeddedConstraint.html) on the displacements, but the input processor now states that it cannot add a constraint for that variable, since it is not a nonlinear variable.

What is the correct way to perfectly bond multiple materials together in MOOSE (as in a composite, for example)?

**Comment by ndaman** [comment, marked as answer]
Well I guess I was just using the wrong constraint, I think I have what I was looking for using an EqualValueBoundaryConstraint, but I'm curious how this is different from a TiedValueConstraint in this case.

**Comment by coskrrb2002** [comment]
I guess you should use Fragments to combine and eliminate any overlapped between two bodies.
I had similar troubles before.
If you use _BooleanFragments_, then there will be no issues!
